packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -181,11 +187,47 @@ bool ArbCore::deliverMessages(
     std::vector<std::vector<unsigned char>> sequencer_batch_items,
     std::vector<std::vector<unsigned char>> delayed_messages,
     const std::optional<uint256_t>& reorg_batch_items) {
-    auto status = message_data_status.load();
-    if (status != MESSAGES_EMPTY) {
-        std::cerr << "unable to deliver messages, status: " << status
-                  << std::endl;
-        return false;
+    uint32_t error_count = 0;
+    auto status_before = MESSAGES_ERROR;
+    while (true) {
+        status_before = message_data_status.load();
+        if (status_before != MESSAGES_EMPTY) {
```

Comment:
I'd recommend also allowing `MESSAGES_SUCCESS` here

Reply:
If we did that, then then `messagesStatus` would never return success.  Each call to `DeliverMessages` needs to be followed with a call to `messagesStatus` to make sure no threads get stuck.

Reply:
Added MESSAGES_SUCCESS here as well as add code to `messagesStatus` to assuming MESSAGES_SUCCESS if MESSAGES_EMPTY is seen.  That breaks the requirement of DeliverMessages/messageStatus pairing, and makes things more asynchronous.